### PR TITLE
Web responsive form line breaks

### PR DESCRIPTION
**Pull request by danieltorres7**
On screens with less than 768px content is limited to 2 columns
On mobile titles have its own line

Depends:
- [x] #513

### CODE REVIEW

**Comment by danieltorres7** [comment]
What do you think @lasley @pedrobaeza @leorochael. Web responsive form views should adapt to better fit smaller screens.

**Comment by danieltorres7** [comment]
Before:
===
captura de tela de 2017-01-02 15-49-54

After:
===
captura de tela de 2017-01-02 15-46-37

File: web_responsive/static/src/less/form_view.less
Comment (pedrobaeza) [inline web_responsive/static/src/less/form_view.less]: Will this collide with _web_sheet_full_width_ module?

Comment (danieltorres7) [reply web_responsive/static/src/less/form_view.less]: Don't know, will test...

**Comment by danieltorres7** [comment]
Just added better screenshots above...

**Comment by pedrobaeza** [comment]
Down arrows from right part are lost on mobile view:

seleccion_009

**Comment by danieltorres7** [comment]
@pedrobaeza, just tested on last commit runbot: http://3276840-514-f788ea.runbot2.odoo-community.org/?db=3276840-514-f788ea-all and got these screenshots (both with down arrows):
captura de tela de 2017-01-02 16-08-09
captura de tela de 2017-01-02 16-07-54

**Comment by danieltorres7** [comment]
@pedrobaeza, where did you see the arrows dissapearing? I'm trying to reproduce that...

**Comment by pedrobaeza** [comment]
I'm using Firefox. That should be the problem. I'm also not seeing that cool buttons for editing or removing the image.

**Comment by pedrobaeza** [comment]
OK, buttons appear at a very low resolution, but that's not useful I think.

File: web_responsive/static/src/less/form_view.less
Comment (danieltorres7) [reply web_responsive/static/src/less/form_view.less]: I've removed `max-width`, should be OK now...

**Comment by danieltorres7** [comment]
> OK, buttons appear at a very low resolution, but that's not useful I think.

This is default Odoo behaviour, I'm not changing that...

**Comment by pedrobaeza** [comment]
OK, I didn't know, but maybe we can change class for showing it at a higher resolution, as mobile views are greater than that I think.

**Comment by danieltorres7** [comment]
> OK, I didn't know, but maybe we can change class for showing it at a higher resolution, as mobile views are greater than that I think.

OK, I'll consider that for another PR. 

BTW, I would prefer that this PR is merged only after rebasing onto #513

**Comment by lasley** [comment]
> BTW, I would prefer that this PR is merged only after rebasing onto #513

To mark that a PR is dependent on another, put this in your original comment with a check list. Something like:

```
Depends:
- [ ] #513
```

Then once it's merged and you rebase, you can just check it off the list 🚀

**Comment by leorochael** [comment]
> OK, I didn't know, but maybe we can change class for showing it at a higher resolution, as mobile views are greater than that I think.

@pedrobaeza: You're right about mobile devices having higher resolution, and the resolution will only get higher with time.

The problem is that it can be argued that what Odoo does here is wrong: it uses screen resolution as a proxy for the absence of a mouse, as it would require having a mouse to hover over the image before seeing the edit icons in high resolution.

In discussions of the design community ([like this](https://css-tricks.com/forums/topic/conditional-css-for-touch-screen-devices/)), the consensus seems to be that hover events should only ever be used to add optional embellishments to the design, but not be necessary for the functionality, no matter the resolution.

So I guess we could have `web_responsive` *fix* Odoo by making the buttons always be present.

But like @danieltorres7 said, this should go into another PR.

**Comment by danieltorres7** [comment]
Since image editing controls are related to forms I just committed an addition to make them always visible.

**Comment by pedrobaeza** [comment]
@Yajo, please review

**Comment by yajo** [comment]
Seems a good IMP. I'll mark as WIP until the other PR is merged and then review 😉

**Comment by leorochael** [comment]
@Yajo, the behaviour of this PR is not really affected by the presence of #513. It just looks prettier when the menus are also better adapted to mobile.

You can review it at your convenience, regardless of #513.

File: web_responsive/static/src/less/form_view.less
Comment (yajo) [inline web_responsive/static/src/less/form_view.less]: Please replace `!important`s for more concise CSS selectors.

File: web_responsive/static/src/less/form_view.less
Comment (yajo) [inline web_responsive/static/src/less/form_view.less]: Replace.

File: web_responsive/static/src/less/form_view.less
Comment (yajo) [inline web_responsive/static/src/less/form_view.less]: Use color variables from where you now know 😉

File: web_responsive/static/src/less/form_view.less
Comment (yajo) [inline web_responsive/static/src/less/form_view.less]: Color variables.

**Review by yajo — CHANGES_REQUESTED** [review]
This seems much better than it was indeed. However it seems incomplete.

See, this is 458×575px window (FF):

captura el 2017-01-04 a las 10 02 23

Do you see the horizontal scroll bar? See what happens when I scroll to the right:

captura el 2017-01-04 a las 10 03 07

I think the problem is in the chatter widget. it gets wider than its parent container. Can you check that please?

**Comment by danieltorres7** [comment]
> I think the problem is in the chatter widget. it gets wider than its parent container. Can you check that please?

Your right, #513 should fix that after merging.

File: web_responsive/static/src/less/form_view.less
Comment (danieltorres7) [reply web_responsive/static/src/less/form_view.less]: This import is to override the base template where td's have width: 100%!important. Precise selector can't do the job.

File: web_responsive/static/src/less/form_view.less
Comment (danieltorres7) [reply web_responsive/static/src/less/form_view.less]: This important is to override the previous important on the same less file.

File: web_responsive/static/src/less/form_view.less
Comment (danieltorres7) [reply web_responsive/static/src/less/form_view.less]: Changing...

File: web_responsive/static/src/less/form_view.less
Comment (danieltorres7) [reply web_responsive/static/src/less/form_view.less]: Changing...

**Comment by mileo** [comment]
@lasley "Other than that though, general rule is one final commit per PR - unless multiple contributors or migrations" Is this recorded in any document written in the OCA? (sorry for the off topic)

**Comment by lasley** [comment]
@mileo [Contribution Docs, Commit guidelines](https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#git). "One logical changeset" typically works out to one per PR, but obviously can vary depending on context - such as multiple authors.

**Comment by pedrobaeza** [comment]
What it still not reflected in guidelines is the agreement we made between several regular contributors for the reviewing process: don't squash commits until the PR is approved. This way, if you have reviewed full PR and make comments, you only need to review again the new commits to see if your comments have been attended. Once the review is done and you approve the PR, that new extra commits like "PEP8", "Requested changes" and so on must be squashed before merging. If all the commits come from the same person and it's only one logical changeset, this can be done using the GitHub feature "Squash and merge". If not, any contributor with write access on the repository can do it (if enabled "Allow to edit to maintainers" on the PR), or the author of the PR otherwise.

This can be included in the guidelines for being clearer if any volunteer.

**Comment by moylop260** [comment]
The best comment about squash commit @pedrobaeza

I can help but my bad english make many typo feedback and you finish writing all again haha

What about create your PR to guidelines?

**Comment by pedrobaeza** [comment]
OK, I'll do it sometime in the day.

File: web_responsive/static/src/less/form_view.less
Comment (yajo) [reply web_responsive/static/src/less/form_view.less]: Same case as https://github.com/OCA/web/pull/513#discussion_r94752243

Comment (leorochael) [reply web_responsive/static/src/less/form_view.less]: Like in #513, there's a `style="width: 100%"` attribute directly on the element (`<td>` in this case) I don't know how to make a selector more specific than the `style` attribute.

We haven't figured out yet where this `style` attribute on the `<td>` is coming from.

Does anyone have any idea?

Comment (leorochael) [reply web_responsive/static/src/less/form_view.less]: Found [the culprit](https://github.com/odoo/odoo/blob/3d085f632d4c5a8aece92731494e1dbbc8923049/addons/web/static/src/js/views/form_view.js#L1499):

     $td.css("width", (nonlabel_width*parseInt($td.attr('colspan')))+"%");

@Yajo what do you suggest?

**Comment by danieltorres7** [comment]
@Yajo, requested changes and final fixes applied. We are ready for the final review.

**Comment by pedrobaeza** [comment]
Dependency merged. Should you rebase?

File: web_responsive/static/src/less/form_view.less
Comment (yajo) [reply web_responsive/static/src/less/form_view.less]: Hmm a tricky one... What about this?

1. Make sure your solution does not break when there are colspans.
2. If it works fine, then I suggest you put an explanation above this line, seems like it would be hard to fix this upstream, and after all they're gonna open the responsive client for v11.

**Review by yajo — CHANGES_REQUESTED** [review]
Rebase, I want to check if nothing breaks now that #513 is merged.

**Comment by leorochael** [comment]
@danieltorres7 is till the end of the week. If he adds me as a developer to his fork I can rebase for him.

**Comment by pedrobaeza** [comment]
I have rebased the branch.

@Yajo, OCA/web maintainers can edit PR branches.

@leorochael, what we do when working collaboratively is to push our PR branches to the team, so anyone in the team can modify them. It's only a suggestion, obviously :smile:

**Review by yajo — CHANGES_REQUESTED** [review]
As you can see below in a screenshot of a partner form, there's still the `.oe_chatter` problem. It has a `min-width` that must be removed for it to not break the layout:

captura de pantalla de 2017-01-17 09-14-38

**Review by yajo — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Please squash the first 4 commits, Daniel.

**Comment by danieltorres7** [comment]
Squashed last 4 commits into 2 commits by relevant changes on each.

Ready for merging in my opinion.

**Review by rafaelbn — APPROVED** [review]
Thanks!

**Pull request merged**
